woocommerce-payments.php

```diff
@@ -98,6 +99,52 @@ function wcpay_init() {
 // If you change the priority of this action, you'll need to change it in the wcpay_check_old_jetpack_version function too.
 add_action( 'plugins_loaded', 'wcpay_init', 11 );
 
+/**
+ * Check WC Payments version and run update related actions.
+ */
+function wcpay_check_version() {
+	if ( defined( 'IFRAME_REQUEST' ) ) {
+		return;
+	}
+
+	$version_option_name = 'wcpay_version';
+
+	$version_option  = get_option( $version_option_name );
+	$requires_update = version_compare( get_option( $version_option_name ), WCPAY_VERSION, '<' );
+
+	if ( ! $version_option || $requires_update ) {
```

Comment:
Actually, there's already a mechanism in place that checks plugin version and handles updates!

It's [hooked](https://github.com/Automattic/woocommerce-payments/blob/0be8aea7400b5947b7fde3bb739ac8448ea270f9/includes/class-wc-payments.php#L145) to `init`. [Here's the method.](https://github.com/Automattic/woocommerce-payments/blob/0be8aea7400b5947b7fde3bb739ac8448ea270f9/includes/class-wc-payments.php#L765)

It checks if the plugin version differs from what's stored in the DB and if it does, calls the `woocommerce_woocommerce_payments_updated` action and then updates the stored plugin version.

Maybe you could hook updating `wcpay_activation_timestamp` to this hook instead?

Reply:
Thank you so much! That's really nice to know! I searched it all over. I should have asked in the slack channel.

Reply:
AFAIR I missed it as well when first working on something related to version updates. 😄

Reply:
@waclawjacek Sorry for the delay in response. I've made a couple of changes to reflect your suggestions. The changes are now simpler 🙇 

Set activation date in the method suggested: https://github.com/Automattic/woocommerce-payments/pull/2313/commits/daa536e563e1464462ebd279c0b4a1c675036960
Enable notification and show it only after 7 days: https://github.com/Automattic/woocommerce-payments/pull/2313/commits/780f548cf2ecbec093211df8a9996b0ef00a4fff

I've also updated testing instructions.

Could you take a look at the changes when you get a chance, please?